File: CS/FilterDropDown_AgregateOperators/Behaviours/FilterDropDownAggregateOperatorBehavior.cs
using DevExpress.Mvvm.UI.Interactivity;
using DevExpress.Xpf.Grid;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;

namespace FilterDropDown_AgregateOperators.Behaviours
{
    internal class FilterDropDownAggregateOperatorBehavior : Behavior<GridControl>
    {


        public string CollectionColumnFieldName
        {
            get { return (string)GetValue(CollectionColumnFieldNameProperty); }
            set { SetValue(CollectionColumnFieldNameProperty, value); }
        }

        
        public static readonly DependencyProperty CollectionColumnFieldNameProperty =
            DependencyProperty.Register("CollectionColumnFieldName", typeof(string), typeof(FilterDropDownAggregateOperatorBehavior), new PropertyMetadata(String.Empty));

        public string DataItemId
        {
            get { return (string)GetValue(DataItemIdProperty); }
            set { SetValue(DataItemIdProperty, value); }
        }

        
        public static readonly DependencyProperty DataItemIdProperty =
            DependencyProperty.Register("DataItemId", typeof(string), typeof(FilterDropDownAggregateOperatorBehavior), new PropertyMetadata(String.Empty));


        public IList<Object> ColumnItemsSource
        {
            get { return (IList<Object>)GetValue(ColumnItemsSourceProperty); }
            set { SetValue(ColumnItemsSourceProperty, value); }
        }
        public static readonly DependencyProperty ColumnItemsSourceProperty =
            DependencyProperty.Register("ColumnItemsSource", typeof(IList<Object>), typeof(FilterDropDownAggregateOperatorBehavior), new PropertyMetadata(null));

        protected override void OnAttached()
        {
            base.OnAttached();
            AssociatedObject.Loaded += AssociatedObject_Loaded;

        }

        private void AssociatedObject_Loaded(object sender, RoutedEventArgs e)
        {
            AssociatedObject.Loaded -= AssociatedObject_Loaded;
            AssociatedObject.SubstituteFilter += AssociatedObject_SubstituteFilter;
            AssociatedObject.View.ShowFilterPopup += View_ShowFilterPopup;
        }

        private void View_ShowFilterPopup(object sender, FilterPopupEventArgs e)
        {
            if (e.Column.FieldName == CollectionColumnFieldName)
            {
                e.ExcelColumnFilterSettings.AllowedFilterTypes = DevExpress.Xpf.Grid.ExcelColumnFilterType.FilterValues;
                e.ExcelColumnFilterSettings.FilterItems = ColumnItemsSource;
            }
        }


        private void AssociatedObject_SubstituteFilter(object sender, DevExpress.Data.SubstituteFilterEventArgs e)
        {
            InToAggregatePatcher.FieldName = CollectionColumnFieldName;
            InToAggregatePatcher.DataItemId = DataItemId;
            e.Filter = InToAggregatePatcher.Patch(e.Filter);
        }

        protected override void OnDetaching()
        {
            AssociatedObject.SubstituteFilter -= AssociatedObject_SubstituteFilter;
            AssociatedObject.View.ShowFilterPopup -= View_ShowFilterPopup;
            base.OnDetaching();
        }


    }
}

File: VB/FilterDropDown_AgregateOperators/Behaviors/FilterDropDownAggregateOperatorBehavior.vb
Imports DevExpress.Mvvm.UI.Interactivity
Imports DevExpress.Xpf.Grid
Imports System.Collections.Generic
Imports System.Windows

Namespace FilterDropDown_AgregateOperators.Behaviors

    Friend Class FilterDropDownAggregateOperatorBehavior
        Inherits Behavior(Of GridControl)

        Public Property CollectionColumnFieldName As String
            Get
                Return CStr(GetValue(CollectionColumnFieldNameProperty))
            End Get

            Set(ByVal value As String)
                SetValue(CollectionColumnFieldNameProperty, value)
            End Set
        End Property

        Public Shared ReadOnly CollectionColumnFieldNameProperty As DependencyProperty = DependencyProperty.Register("CollectionColumnFieldName", GetType(String), GetType(FilterDropDownAggregateOperatorBehavior), New PropertyMetadata(String.Empty))

        Public Property DataItemId As String
            Get
                Return CStr(GetValue(DataItemIdProperty))
            End Get

            Set(ByVal value As String)
                SetValue(DataItemIdProperty, value)
            End Set
        End Property

        Public Shared ReadOnly DataItemIdProperty As DependencyProperty = DependencyProperty.Register("DataItemId", GetType(String), GetType(FilterDropDownAggregateOperatorBehavior), New PropertyMetadata(String.Empty))

        Public Property ColumnItemsSource As IList(Of Object)
            Get
                Return CType(GetValue(ColumnItemsSourceProperty), IList(Of Object))
            End Get

            Set(ByVal value As IList(Of Object))
                SetValue(ColumnItemsSourceProperty, value)
            End Set
        End Property

        Public Shared ReadOnly ColumnItemsSourceProperty As DependencyProperty = DependencyProperty.Register("ColumnItemsSource", GetType(IList(Of Object)), GetType(FilterDropDownAggregateOperatorBehavior), New PropertyMetadata(CType(Nothing, PropertyChangedCallback)))

        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            AddHandler AssociatedObject.Loaded, AddressOf AssociatedObject_Loaded
        End Sub

        Private Sub AssociatedObject_Loaded(ByVal sender As Object, ByVal e As RoutedEventArgs)
            RemoveHandler AssociatedObject.Loaded, AddressOf AssociatedObject_Loaded
            AddHandler AssociatedObject.SubstituteFilter, AddressOf AssociatedObject_SubstituteFilter
            AddHandler AssociatedObject.View.ShowFilterPopup, AddressOf View_ShowFilterPopup
        End Sub

        Private Sub View_ShowFilterPopup(ByVal sender As Object, ByVal e As FilterPopupEventArgs)
            If Equals(e.Column.FieldName, CollectionColumnFieldName) Then
                e.ExcelColumnFilterSettings.AllowedFilterTypes = ExcelColumnFilterType.FilterValues
                e.ExcelColumnFilterSettings.FilterItems = ColumnItemsSource
            End If
        End Sub

        Private Sub AssociatedObject_SubstituteFilter(ByVal sender As Object, ByVal e As DevExpress.Data.SubstituteFilterEventArgs)
            InToAggregatePatcher.FieldName = CollectionColumnFieldName
            InToAggregatePatcher.DataItemId = DataItemId
            e.Filter = InToAggregatePatcher.Patch(e.Filter)
        End Sub

        Protected Overrides Sub OnDetaching()
            RemoveHandler AssociatedObject.SubstituteFilter, AddressOf AssociatedObject_SubstituteFilter
            RemoveHandler AssociatedObject.View.ShowFilterPopup, AddressOf View_ShowFilterPopup
            MyBase.OnDetaching()
        End Sub
    End Class
End Namespace
